### ISSUE TITLE
destroy / reinit

### ISSUE THREAD

**Issue by jasonalvis** [Strategery-Inc/Magento-InfiniteScroll#123]
Hello,

Has anyone worked out a solution to re-initialising the infinite scroll, for example when using the layered navigation an ajax call is made, I need to be able to re-init the infinite scroll on ajax success but not sure what I need to call to achieve this.

I've tried numerous things but none of them work.

    $(document).bind("m-ajax-after", function() {
        // re-init the infinite scroll    
    });

**Comment by gabrielsomoza** [comment]
Thanks for opening the ticket. We didn't have to tackle this problem with any of our clients yet so we don't have a solution at the moment. But I'll of course leave the ticket open, maybe someone in the community did solve it.

@damian-pastorini: please try to you circle back to this ticket whenever you get a chance in your busy schedule.

**Comment by damian-pastorini** [comment]
Hi @jasonalvis , you can try with the following:

```$(document).bind("m-ajax-after", function() {
   SgyIAS.init(); // or window.SgyIAS.init();
});```

If you already loaded the infinitscroll on the page the SgyIAS should be already created.

**Comment by phpsos** [comment]
hey!
great extension, everything works, except ajax loading...
@damian-pastorini 
could you please check out if this code works.for you?
I tried it and this function called is unedfined. Even though before ajax load, infinity scroll works as it should.

**Comment by Vigneshmuthusamy** [comment]
I'm also trying to integrate manadev layered navigation filter with this plugin. But after ajax filter is applied, it doesn't trigger inifinite scroll. Instead it shows normal pagination.

I've tried the above solution too. But still it isn't working. Any suggestions?

**Comment by gsomoza** [comment]
We can't provide suggestions unless we understand what the issues are once you try to re-init. 
* There must be some kind of error happening,
* Or maybe nothing happens and things silently stop working?

I suggest starting a new branch forked from this repo where we can all look at what you did so far and based on that run our own tests, etc. Would any of you be willing to share that with us?

**Comment by damian-pastorini** [comment]
@gsomoza as far I understand the issue is related to a 3rd party extension to make the Layer Navigation work though AJAX.
@phpsos , @Vigneshmuthusamy , I'm sorry but we can't give you free support on this, more than a few recommendations.

In this particular case you should make sure that the bind is working, if the commented line:
jQuery(document).bind("m-ajax-after", function() { alert("test");  });
doesn't work, you can try by run it inside a document.ready event or something like that. 
Is possible that the ajax event (m-ajax-after) is been declared after your bind declaration if you add it at the end of our template.

In summary, if the bind doesn't work you won't be able to re-init the InfiniteScroll.

If you like you can hire our support services and we will be glad to help you, we will also need access to your code.

Best,

**Comment by damian-pastorini** [comment]
If you need dev support please contact us at contact (at) strategery (dot) io

**Comment by DevertNet** [comment]
This works for me with the ajax filter of "Amasty_ImprovedLayeredNavigation".
fyi: I must rewrite the function, because i didn't find any callbacks in "amshopby_ajax_update".
**Important** You must update Infinite Ajax Scroll to v2.2.1 or up for the reinitialize function!
```js
if(typeof(window.amshopby_ajax_update)!='undefined'){
    var amshopby_ajax_update_old = window.amshopby_ajax_update;
    window.amshopby_ajax_update = function(data){
        amshopby_ajax_update_old(data);
        if(typeof(window.ias)!='undefined') window.ias.reinitialize();
    }
}
```

**Comment by gsomoza** [comment]
Thanks a lot for sharing this with the community!
